library/src/main/java/com/wultra/android/mtokensdk/common/Logger.kt

```diff
@@ -0,0 +1,122 @@
+ * This source code can be used only by the owner of the license.
+ *
+ * Any disputes arising in respect of this agreement (license) shall be brought
+ * before the Municipal Court of Prague.
+ */
+
+package com.wultra.android.mtokensdk.common
+
+import android.util.Log
+import okhttp3.Headers
+import okhttp3.OkHttpClient
+import okio.Buffer
+import java.lang.StringBuilder
+
+/**
+ * Logger provides simple logging facility.
+ *
+ * Logs are written with "WMT" tag to standard [android.util.Log] logger.
+ */
+class Logger {
+
+    enum class VerboseLevel {
+        /** Silences all messages. */
+        OFF,
+        /** Only errors will be printed into the log. */
+        ERROR,
+        /** Errors and warnings will be printed into the log. */
+        WARNING,
+        /** All messages will be printed into the log. */
+        DEBUG
+    }
+
+    companion object {
+
+        @JvmStatic
+        /** Current verbose level. */
+        var verboseLevel = VerboseLevel.WARNING
+
+        private val tag = "WMT"
+
+        internal fun d(message: String) {
+            if (verboseLevel.ordinal >= VerboseLevel.DEBUG.ordinal) {
+                Log.d(tag, message)
+            }
+        }
+
+        internal fun d(fn: () -> String) {
+            if (verboseLevel.ordinal >= VerboseLevel.DEBUG.ordinal) {
+                Log.d(tag, fn())
+            }
+        }
+
+        internal fun w(message: String) {
+            if (verboseLevel.ordinal >= VerboseLevel.WARNING.ordinal) {
+                Log.w(tag, message)
+            }
+        }
+
+        internal fun w(fn: () -> String) {
+            if (verboseLevel.ordinal >= VerboseLevel.WARNING.ordinal) {
+                Log.w(tag, fn())
+            }
+        }
+
+        internal fun e(message: String, t: Throwable? = null) {
+            if (verboseLevel.ordinal >= VerboseLevel.ERROR.ordinal) {
+                Log.e(tag, message, t)
+            }
+        }
+
+        internal fun e(fn: () -> String) {
+            if (verboseLevel.ordinal >= VerboseLevel.ERROR.ordinal) {
+                Log.e(tag, fn())
+            }
+        }
+
+        internal fun configure(builder: OkHttpClient.Builder) {
+            builder.addInterceptor { chain ->
+                val request = chain.request()
+                d {
+                    var body = ""
+                    try {
+                        val buffer = Buffer()
+                        request.newBuilder().build().body()?.writeTo(buffer)
+                        body = buffer.readUtf8()
+                    } catch (e: Throwable) {
+                        e("Failed to parse request body")
+                    }
+
+                    "\n--- WMT REQUEST ---" +
+                    "\n- URL: ${request.method()} - ${request.url()}" +
+                    "\n- Headers: ${request.headers().forLog()}" +
+                    "\n- Body: $body"
+                }
+                val response = chain.proceed(request)
```

Comment:
Method `proceed` can throw an Exception.
If you check `HttpLoggingInterceptoor`, you find this:
```
try {
      response = chain.proceed(request)
} catch (e: Exception) {
      logger.log("<-- HTTP FAILED: $e")
      throw e
}
```
It might be a good idea to log the exception too.

Reply:
Good idea, fixed